mass-deployer/pkg/mass-deployer/deployer.go

```diff
@@ -284,3 +293,56 @@ func parseDisks(name string, disks []Disk) (disksWorkloads []workloads.Disk, mou
 	}
 	return
 }
+
+func updateFailedDeployments(tfPluginClient deployer.TFPluginClient, nodesIDs []int, groupDeployments *groupDeploymentsInfo) {
+	// update nodes for failed networks and update vm deployments accordingly
+	// return if the networks were updated
+	var updated bool
+	for idx, deployment := range groupDeployments.networkDeployments {
+
+		nodeID := uint32(nodesIDs[idx%len(nodesIDs)])
+		if isFailedNetwork(*deployment) {
```

Comment:
`isFailedNetwork` returns true if one contract is 0. can some contracts still exist with the 0 contract? if that's the case we will need to cancel these contracts too.

Reply:
it's not the case with mass deployer, the network has one node in every network

Reply:
it still could deploy more than one in public node example